tools/unicode_properties_parse/format_width_estimate_intervals.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// The following code generates data for `_Width_estimate_intervals_v2` in <format>.
+
+#include <charconv>
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <string>
+#include <system_error>
+#include <vector>
+using namespace std;
+
+void verify_impl(bool test, int line, const char* msg) {
+    if (!test) {
+        cerr << "Error at line " << line << ": " << msg << endl;
+        exit(EXIT_FAILURE);
+    }
+}
+#define VERIFY(expr, msg) verify_impl((expr), __LINE__, (msg))

```

Comment:
```suggestion
void verify(bool test, const char* msg, source_location loc = source_location::current()) {
    if (!test) {
        cerr << "Error at line " << loc.line() << ": " << msg << endl;
        exit(EXIT_FAILURE);
    }
}
```

> `VERIFY(regex_match(line, match, reg), "invalid line");`

After second thought, I think a pure function should be better. The macro-based approach gives me an unreal feeling that this line can be wrongly omitted in certain cases.